diff --git a/pyhook/CMakeLists.txt b/pyhook/CMakeLists.txt
--- a/pyhook/CMakeLists.txt
+++ b/pyhook/CMakeLists.txt
@@ -39,7 +39,19 @@ else()
       set(_PYHOOK_SYSCONFIG_PREFIX " 'posix_prefix',")
     endif()
     execute_process(
-      COMMAND "${Python3_EXECUTABLE}" -c "import sysconfig as sc; print(sc.get_path('platlib',${_PYHOOK_SYSCONFIG_PREFIX} vars={'platbase': '.'}))"
+      COMMAND "${Python3_EXECUTABLE}" -c "
+import sys
+import sysconfig
+import os
+
+platlib = sysconfig.get_path('platlib')
+prefix = sys.prefix
+
+if platlib.startswith(prefix):
+    print(platlib[len(prefix):].lstrip(os.sep))
+else:
+    print(platlib)
+"
       RESULT_VARIABLE _pyhook_install_dir_result
       OUTPUT_VARIABLE PYHOOK_INSTALL_DIR
       OUTPUT_STRIP_TRAILING_WHITESPACE)
